libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -126,14 +124,12 @@ uint8_t AP_ESC_Telem::get_max_rpm_esc() const
     uint32_t ret = 0;
     float max_rpm = 0;
     const uint32_t now = AP_HAL::millis();
-    const uint32_t now_us = AP_HAL::micros();
     for (uint8_t i = 0; i < ESC_TELEM_MAX_ESCS; i++) {
         if (_telem_data[i].last_update_ms == 0 && !was_rpm_data_ever_reported(_rpm_data[i])) {
             // have never seen telem from this ESC
             continue;
         }
-        if (_telem_data[i].stale(now)
-            && !rpm_data_within_timeout(_rpm_data[i], now_us, ESC_RPM_DATA_TIMEOUT_US)) {
+        if (_telem_data[i].stale(now) && !_rpm_data[i].data_valid) {
```

Comment:
```suggestion
        if (_telem_data[i].stale(now) || !_rpm_data[i].data_valid) {
```
If data stale or not valid then don't used it?